pkg/ocm/managedcluster.go

```diff
@@ -0,0 +1,219 @@
+type ManagedClusterAdditionalOptions func(builder *ManagedClusterBuilder) (*ManagedClusterBuilder, error)
+
+// NewManagedClusterBuilder creates a new instance of ManagedClusterBuilder.
+func NewManagedClusterBuilder(apiClient *clients.Settings, name string) *ManagedClusterBuilder {
+	glog.V(100).Infof(
+		`Initializing new ManagedCluster structure with the following params: name: %s`, name)
+
+	builder := ManagedClusterBuilder{
+		apiClient: &apiClient.ClusterV1Client,
+		Definition: &mcV1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: mcV1.ManagedClusterSpec{},
+		},
+	}
+
+	if builder.apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		builder.errorMsg = "managedcluster 'apiClient' cannot be empty"
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the ManagedCluster is empty")
+
+		builder.errorMsg = "ManagedCluster 'name' cannot be empty"
+	}
+
+	return &builder
+}
+
+// WithOptions creates ManagedCluster with generic mutation options.
+func (builder *ManagedClusterBuilder) WithOptions(
+	options ...ManagedClusterAdditionalOptions) *ManagedClusterBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting ManagedCluster additional options")
+
+	for _, option := range options {
+		if option != nil {
+			builder, err := option(builder)
+
+			if err != nil {
+				glog.V(100).Infof("Error occurred in mutation function")
+
+				builder.errorMsg = err.Error()
+
+				return builder
+			}
+		}
+	}
+
+	return builder
+}
+
+// PullManagedCluster loads an existing ManagedCluster into ManagedClusterBuilder struct.
+func PullManagedCluster(apiClient *clients.Settings, name string) (*ManagedClusterBuilder, error) {
+	glog.V(100).Infof("Pulling existing ManagedCluster name: %s", name)
+
+	builder := ManagedClusterBuilder{
+		apiClient: &apiClient.ClusterV1Client,
+		Definition: &mcV1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		builder.errorMsg = "managedcluster 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("managedcluster object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches the defined ManagedCluster from the cluster.
+func (builder *ManagedClusterBuilder) Get() (*mcV1.ManagedCluster, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting ManagedCluster %s", builder.Definition.Name)
+
+	ManagedCluster := &mcV1.ManagedCluster{}
+	ManagedCluster, err := builder.apiClient.ManagedClusters().Get(context.TODO(), builder.Definition.Name, metav1.GetOptions{})
+
+	// err := builder.apiClient.Get(context.TODO(), runtimeClient.ObjectKey{
+	// 	Name: builder.Definition.Name,
+	// }, ManagedCluster)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return ManagedCluster, err
+}
```

Comment:
Since the API client already provides a Get method, you can remove this

Reply:
Fixed